# Primer Avance - PIA

<b>Equipo: </b> 1 <br> 
<b>Grupo: </b> 002, Miércoles 18:00-21:00 <br> 
<b>Integrantes: </b> 
 
- 1794656 Félix Barreras Jesús Guillermo
- 1723237 Linares Núñez Vanessa Stefanía
- 1749744 Ramirez Vargas Dayan
- 1803744 Rodríguez Estrada Erick Alejandro

### Título de la Base de Datos 

a) Nombre con el que se encuentra en la página encontrada: 
“World Happiness Report”

b) Url de la página: 
https://www.kaggle.com/unsdsn/world-happiness?select=2019.csv


### Descripción de los Datos

a) ¿Qué tipo de datos son?

El Informe Mundial sobre la Felicidad es un estudio histórico sobre el estado de la felicidad mundial. Las puntuaciones y clasificaciones de la felicidad utilizan datos de la Encuesta Mundial de Gallup. Las puntuaciones se basan en las respuestas a la principal pregunta de evaluación de la vida formulada en la encuesta.

b) Descripción de las columnas:
1. Calificación general: numérico (entero)
2. País o Región: descriptivo
3. Puntaje: numérico(flotante)
4. Economía (PIB per cápita): numérico (flotante)
5. Apoyo social: numérico (flotante)
6. Salud (esperanza de vida): numérico (flotante)
7. Libertad para tomar decisiones de vida: numérico (flotante)
8. Generosidad: numérico (flotante)
9. Percepciones de corrupción: numérico(flotante)


### Justificación del Uso de Datos

a) Características atractivas de los datos

Los datos presentan, como característica más importante para el equipo, ser internacionales, pues se incluyen 156 países. Es una base de datos lanzada por la Organización de las Naciones Unidas y tiene utilidad en una gran variedad de campos como lo son la economía, la psicología, la salud, entre muchas otras más. 

b) Beneficios de trabajar con estos datos

La base de datos cuenta con una gran variedad de datos numéricos, por lo que creemos que da oportunidad a ser analizada y trabajada con bastante libertad, así como dar respuesta a múltiples preguntas, según sea el caso. Además, el hecho de contener a una gran cantidad de países, cualquier proyecto que resulte de ella, puede ser de interés internacional. 

### Objetivo Final 

a) Explicación de los objetivos (principal y secundarios)

 - Objetivo principal: Esta base de datos nos muestra los 156 países más felices del mundo, basándose en diferentes factores que podrían influir en la felicidad de los habitantes de estos (libertad, corrupción, economía, etc.) Al trabajar con esta base de datos podemos encontrar la respuesta a la pregunta: ¿cuáles son aquellos factores que verdaderamente contribuyen a la felicidad?
 - Objetivos secundarios:Entre los objetivos secundarios se tienen; identificar a los países con la población más feliz, aportar sugerencias para ser más feliz y resaltar los factores a los que se le tienen que dar más importancia en los gobiernos. 


### Preguntas de Interés

a)¿Cuáles son aquellos factores que verdaderamente contribuyen a la felicidad?

b)¿Qué acciones puede ejercer un gobierno para tener una población más feliz?

c)¿A qué otros factores contribuye ser feliz?


### Práctica 1: Limpieza de Datos

PREPARACIÓN DE DATOS
1. Importar librerías

In [1]:
import pandas as pd
import numpy as np
import json

2. Importar base de datos:

In [2]:
df = pd.read_csv("uncleandata_happiness_2019.csv")

Recordar que para que el archivo csv pueda ser leído apropiadamente se debe de encontrar en la misma carpeta(direccion) que nuestro archivo de python

### Visualización de los Datos

In [3]:
df

Overall rank         Country or region  Score  GDP per capita  \
0               1                   Finland  7.769           1.340   
1               2                   Denmark  7.600           1.383   
2               3                    Norway  7.554           1.488   
3               4                   Iceland  7.494           1.380   
4               5               Netherlands  7.488           1.396   
..            ...                       ...    ...             ...   
151           152                    Rwanda  3.334           0.359   
152           153                  Tanzania  3.231           0.476   
153           154               Afghanistan  3.203           0.350   
154           155  Central African Republic  3.083           0.026   
155           156               South Sudan  2.853           0.306   

     Social support  Healthy life expectancy  Freedom to make life choices  \
0             1.587                    0.986                         0.596   
1             1.573                    0.996                         0.592   
2             1.582                    1.028                         0.603   
3             1.624                    1.026                         0.591   
4             1.522                    0.999                         0.557   
..              ...                      ...                           ...   
151           0.711                    0.614                         0.555   
152           0.885                    0.499                         0.417   
153           0.517                    0.361                         0.000   
154           0.000                    0.105                         0.225   
155           0.575                    0.295                         0.010   

     Generosity  Perceptions of corruption  
0         0.153                      0.393  
1         0.252                      0.410  
2         0.271                      0.341  
3         0.354                      0.118  
4         0.322                      0.298  
..          ...                        ...  
151       0.217                      0.411  
152       0.276                      0.147  
153       0.158                      0.025  
154       0.235                      0.035  
155       0.202                      0.091  

[156 rows x 9 columns]

1. Revisar cuales son los datos que conforman la base de datos:

In [4]:
df.shape
print("Numero de filas:", df.shape[0], "\nNumero de columnas:", df.shape[1])

Numero de filas: 156 
Numero de columnas: 9


2. Ver los nombres de las columnas de nuestra base de datos

In [5]:
df.columns

Index(['Overall rank', 'Country or region', 'Score', 'GDP per capita',
       'Social support', 'Healthy life expectancy',
       'Freedom to make life choices', 'Generosity',
       'Perceptions of corruption'],
      dtype='object')

3. Cambiar los nombres de las columnas a uno que nos facilite la extracción de su información

In [6]:
df = df.rename(columns={'Overall rank': 'Clasificacion_General', 
                              'Country or region':'Pais_o_region',
                              'Score':'Puntuacion',
                              'GDP per capita': 'PIB_per_capita',
                              'Social support':'Apoyo_Social',
                              'Healthy life expectancy':'Esperanza_de_vida', 
                              'Freedom to make life choices':'Libertad',
                              'Generosity':'Generocidad',
                              'Perceptions of corruption':'Percepción_de_corrupcion'})

df

Clasificacion_General             Pais_o_region  Puntuacion  \
0                        1                   Finland       7.769   
1                        2                   Denmark       7.600   
2                        3                    Norway       7.554   
3                        4                   Iceland       7.494   
4                        5               Netherlands       7.488   
..                     ...                       ...         ...   
151                    152                    Rwanda       3.334   
152                    153                  Tanzania       3.231   
153                    154               Afghanistan       3.203   
154                    155  Central African Republic       3.083   
155                    156               South Sudan       2.853   

     PIB_per_capita  Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0             1.340         1.587              0.986     0.596        0.153   
1             1.383         1.573              0.996     0.592        0.252   
2             1.488         1.582              1.028     0.603        0.271   
3             1.380         1.624              1.026     0.591        0.354   
4             1.396         1.522              0.999     0.557        0.322   
..              ...           ...                ...       ...          ...   
151           0.359         0.711              0.614     0.555        0.217   
152           0.476         0.885              0.499     0.417        0.276   
153           0.350         0.517              0.361     0.000        0.158   
154           0.026         0.000              0.105     0.225        0.235   
155           0.306         0.575              0.295     0.010        0.202   

     Percepción_de_corrupcion  
0                       0.393  
1                       0.410  
2                       0.341  
3                       0.118  
4                       0.298  
..                        ...  
151                     0.411  
152                     0.147  
153                     0.025  
154                     0.035  
155                     0.091  

[156 rows x 9 columns]

4. Conocer el tipo de dato que contiene cada columna

In [7]:
df.dtypes

Clasificacion_General         int64
Pais_o_region                object
Puntuacion                  float64
PIB_per_capita              float64
Apoyo_Social                float64
Esperanza_de_vida           float64
Libertad                    float64
Generocidad                 float64
Percepción_de_corrupcion    float64
dtype: object

### Descripción Básica de las columnas

In [8]:
df

Clasificacion_General             Pais_o_region  Puntuacion  \
0                        1                   Finland       7.769   
1                        2                   Denmark       7.600   
2                        3                    Norway       7.554   
3                        4                   Iceland       7.494   
4                        5               Netherlands       7.488   
..                     ...                       ...         ...   
151                    152                    Rwanda       3.334   
152                    153                  Tanzania       3.231   
153                    154               Afghanistan       3.203   
154                    155  Central African Republic       3.083   
155                    156               South Sudan       2.853   

     PIB_per_capita  Apoyo_Social  Esperanza_de_vida  Libertad  Generocidad  \
0             1.340         1.587              0.986     0.596        0.153   
1             1.383         1.573              0.996     0.592        0.252   
2             1.488         1.582              1.028     0.603        0.271   
3             1.380         1.624              1.026     0.591        0.354   
4             1.396         1.522              0.999     0.557        0.322   
..              ...           ...                ...       ...          ...   
151           0.359         0.711              0.614     0.555        0.217   
152           0.476         0.885              0.499     0.417        0.276   
153           0.350         0.517              0.361     0.000        0.158   
154           0.026         0.000              0.105     0.225        0.235   
155           0.306         0.575              0.295     0.010        0.202   

     Percepción_de_corrupcion  
0                       0.393  
1                       0.410  
2                       0.341  
3                       0.118  
4                       0.298  
..                        ...  
151                     0.411  
152                     0.147  
153                     0.025  
154                     0.035  
155                     0.091  

[156 rows x 9 columns]

<b>Descripción: </b> 

1. Clasificación General:

La columna de clasificación general indica el ranking de cada país, siendo el número 1 el más feliz de la muestra y el 156 el menos. Es de caracter númerico y va de 1 al 156 sin repetir. 

2. País o región:

La columna de país o región indica el nombre del país que esta siendo evaluado en el idioma inglés.

3. Puntuación:

La suma de todos los factores siguientes da como resultado el puntaje de felicidad, que es el que se contiene en la columna de puntuación. Son número flotantes que van del 2.853 al 7.769.

4. PIB per cápita, Apoyo Social, Esperanza de vida, Libertad, Generocidad, Percepcion de corrupción:

Para explicar estas columnas primero es importante definir la distopía. 

>La distopía es un país imaginario que tiene a las personas menos felices del mundo. El propósito de establecer Distopía es tener un punto de referencia con el cual todos los países puedan ser comparados favorablemente (ningún país se desempeña peor que Distopía) en términos de cada una de las seis variables clave, permitiendo así que cada sub-barra tenga un ancho positivo. Las puntuaciones más bajas observadas para las seis variables clave, por lo tanto, caracterizan a la distopía. Dado que la vida sería muy desagradable en un país con los ingresos más bajos del mundo, la menor esperanza de vida, la menor generosidad, la mayor parte de la corrupción, la menor libertad y el menor apoyo social, se lo conoce como "distopía", en contraste con la utopía.

Ahora bien, las columnas: PIB per cápita, Apoyo Social, Esperanza de vida, Libertad, Generosidad y Percepción de Corrupción describen en qué medida estos factores contribuyen a evaluar la felicidad en cada país tomando como punto de comparación los valores otorgados a Distopía. 

### Limpieza de datos

>En nuestro caso, como el objetivo de nuestro trabajo es analizar cuales son aquellos factores que influyen en la felicidad de las personas, hemos decidido que no es necesario eliminar ninguna columna. Esto ya que cada columna representa los diversos factores que analizaremos. Así mismo no existen datos nulos.

### Crear un nuevo archivo con la base de datos editada

In [9]:
df.to_csv('newdata_happiness.csv')